### PR TITLE
added RETRY_COUNT environment variable

### LINKED ISSUE #35225: Allow configuring retries for JavaScript tests using environment variable
### Description

The main workflow allows configuring number of retries through an environment variable; https://github.com/keycloak/keycloak/blob/main/.github/workflows/ci.yml#L13.

The Cypress and Playright tests hardcodes this:

* https://github.com/keycloak/keycloak/blob/main/js/apps/admin-ui/cypress.config.ts#L19
* https://github.com/keycloak/keycloak/blob/main/js/apps/account-ui/playwright.config.ts#L12

It would be good to have an environment variable defined in https://github.com/keycloak/keycloak/blob/main/.github/workflows/js-ci.yml#L12 where the number of retries are set.

It should be possible to set the value to `0` for a test to run only 1 time, or `1` for a test to be retried one time.

This will make it easy to configure in the workflow, but also make it easier to schedule a large number of runs without retries to access the stability of the testsuite, as when retries are enabled it is a lot harder to uncover flaky tests.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by edewit**
fixes: #35225
Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>

### CODE REVIEW

**Comment by stianst** [comment]
LGTM, and thanks for doing this one :)

File: js/apps/admin-ui/cypress.config.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import fs from "node:fs";
 import { isAsyncFunction } from "node:util/types";
 
 const isCI = process.env.CI === "true";
+const retryCount = parseInt(process.env.RETRY_COUNT || "3");
```
[the remark this answers is not in the document]
Comment (edewit) [reply js/apps/admin-ui/cypress.config.ts:7]: Because it was like this before

[the remark this answers is not in the document]
Comment (jonkoops) [reply js/apps/admin-ui/cypress.config.ts:7]: Perhaps the default should be changed to `0` here as well?

**Review by jonkoops — DISMISSED** [review]

**Review by agagancarczyk — DISMISSED** [review]

File: js/apps/admin-ui/cypress.config.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (stianst) [reply js/apps/admin-ui/cypress.config.ts:7]: +1 To not enable retries by default

[the remark this answers is not in the document]
Comment (edewit) [reply js/apps/admin-ui/cypress.config.ts:7]: We can change it when our tests are more stable let's leave it like this for now

[the remark this answers is not in the document]
Comment (jonkoops) [reply js/apps/admin-ui/cypress.config.ts:7]: I think @stianst means to set the default to `0` and set the `RETRY_COUNT` value to `3` in CI for now.

Comment (stianst) [inline js/apps/admin-ui/cypress.config.ts:7]:
```suggestion
const retryCount = parseInt(process.env.RETRY_COUNT || "0");
```

**Review by jonkoops — APPROVED** [review]

**Review by ssilvert — APPROVED** [review]

**Pull request merged**